Formula/scotch.rb

```diff
@@ -4,40 +4,42 @@ class Scotch < Formula
   url "https://gitlab.inria.fr/scotch/scotch/-/archive/v6.1.1/scotch-v6.1.1.tar.bz2"
   sha256 "543ab2f8998658e5a6123231bfb3736cfbb939f330c656f8a7913d967a933f00"
   license "CECILL-C"
+  revision 1
   head "https://gitlab.inria.fr/scotch/scotch.git", branch: "master"
 
   livecheck do
     url :stable
     regex(/^v?(\d+(?:\.\d+)+)$/i)
   end
 
-  bottle do
-    sha256 cellar: :any_skip_relocation, arm64_big_sur: "a2a88eca7948940c07bc2263143b35be2d0a6d4f66bf3f4bff93ead83a653921"
-    sha256 cellar: :any_skip_relocation, monterey:      "63ef5a05bb51ff7adaafb4633d09ac262c113341d4b8607460e30521de81d0ff"
-    sha256 cellar: :any_skip_relocation, big_sur:       "54e1182e94e87f15ba4671dd4611efd9aaf6dfa4c480fc304f40633032a70a19"
-    sha256 cellar: :any_skip_relocation, catalina:      "47642fe1c6bfc3da89aa509ac9220d9e27071e30327fb6c607e6473d4cc4d7d4"
-    sha256 cellar: :any_skip_relocation, mojave:        "ec79906f2bc0c47ac0989a7d4dcc6a3a7a49021e8a330b1f04a973ed09d01b0e"
-    sha256 cellar: :any_skip_relocation, x86_64_linux:  "f93fe6e8bcdb31271f7d082915061d6fa2b9e415611a6f52929dbb45a48dd847"
-  end
-
   depends_on "open-mpi" => :build
 
   uses_from_macos "bison"
   uses_from_macos "flex"
   uses_from_macos "zlib"
 
   def install
-    cd "src"
-    (buildpath/"src").install_symlink "Make.inc/Makefile.inc.i686_mac_darwin10" => "Makefile.inc"
-    system "make"
-    system "make", "prefix=#{prefix}", "install"
+    chdir "src" do
+      (buildpath/"src").install_symlink "Make.inc/Makefile.inc.i686_mac_darwin10" => "Makefile.inc"
+
+      inreplace "Makefile.inc" do |s|
+        s.change_make_var! "CCS", ENV.cc
+        s.change_make_var! "CCP", "mpicc"
+        s.change_make_var! "CCD", "mpicc"
+        # If you have doubts on the stability of your MPI implementation, it is better not to set "-DSCOTCH_PTHREAD".
+        s.gsub! "-DSCOTCH_PTHREAD", ""
```

Comment:
Why do we have doubts about this?

Reply:
I copied this line from developer’s installation notes. For best compatibility it shouldn’t be enabled.

Reply:
I copied this line from developer’s installation notes. For best compatibility it shouldn’t be enabled.

Reply:
The comment seems to apply to old MPI implementations. Are you able to demonstrate reproducible issues with this set? If not, we should use the default.

Reply:
You are right. The project I'm running doesn't use `MPI_Init()`, so I cannot reproduce any issue here.
However, other Linux distros choose better compatibility.
RedHat removed `-DSCOTCH_PTHREAD` https://bugzilla.redhat.com/show_bug.cgi?id=1386707
Debian also removed `-DSCOTCH_PTHREAD` https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=612621

Reply:
I'm inclined to leave it unless we get complaints too. Those issues are from 2011 and 2016, I hope things got better since then

Reply:
Fine by me. Mac users don't care about backward compatibility anyway.

Reply:
Keeping `-DSCOTCH_PTHREAD`. I don't see any formula depends on `scotch` so there shouldn't be any problem at least within homebrew.